### ISSUE TITLE
autoencoders

### ISSUE THREAD

**Issue by renukachittimalla** [keras-team/keras#9433]
Hello..I have a situation where i have very less number of labeled examples and lots of unlabeled examples.I am using autoencoder model to train the unlabeled examples. Now i wanted to train encoder with labels. My doubt here is by compiling the encoder model am i creating a new model and not reusing the weights of the autoencoder.Here is the Encoder model a separate CNN model or is still part of autoencoder? 

x = Input(shape=(input_dim,))
encoder = Dense(output_dim)(x)
decoder = Dense(input_dim)(encoder)
 
autoencoder = Model(input=x, output=decoder)

out=dense(10,activation='softmax')(encoder)
supervised = Model(input=x, output=out)
 
autoencoder.compile(...)
supervised.compile(...)

autoencoder.fit(x1,x1)
supervised.fit(x2,y2)

supervised.predict(x3)

Thanks in advance

**Comment by isaacgerg** [comment]
This really isnt a problem with keras but more of a teaching question.  Can you please close?
